### PR TITLE
[WIP] expose disableSNI setting, update docs a bit

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

**Comment by ktoso** [comment]
W8, not complete yet.

**Comment by ktoso** [comment]
Please review @johanandren 


@wsargent  Yeah, no joy here but we're very loud at logging warnings and will document how to minimise damage (do this for a single connection in Akka HTTP)

File: README.md [hunk 1]
```diff
@@ -12,6 +12,19 @@ implementations and/or defining additional settings for the Java runtime to use
...
+
+ssl-config at this point in time is used primarily internally in Akka HTTP, and is being evolved
+towards being "the place" one configures all possible SSL/TLS related settings, mostly focused on 
+the client side of things. 
+
+The project is hosted externally of either Akka or Play, in order to foster convergence and re-use
+of the more tricky bits of configuring TLS.
+
+Binary compatibility is **not guaranteed** between versions (in the `0.x.z` series) of ssl-config at this point in time.
+We aim to stabilise the APIs and provide a stable release eventually.
```
Comment (johanandren) [inline README.md]: Good to be explicit about it

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
Turns out we can't to the SNI things "just so" because JDK7 – does anyone care about JDK7 in ssl-config? // cc @wsargent 
For us it'd be ok to move on to 8, since that Akka 2.4 requires anyway.

Failure was:

```
[error] /home/travis/build/typesafehub/ssl-config/ssl-config-akka/src/main/scala/com/typesafe/sslconfig/akka/AkkaSSLConfig.scala:188: value setServerNames is not a member of javax.net.ssl.SSLParameters
[error]     defaultParams.setServerNames(Collections.emptyList())
[error]                   ^
[error] /home/travis/build/typesafehub/ssl-config/ssl-config-akka/src/main/scala/com/typesafe/sslconfig/akka/AkkaSSLConfig.scala:189: value setSNIMatchers is not a member of javax.net.ssl.SSLParameters
[error]     defaultParams.setSNIMatchers(Collections.emptyList())
[error]                   ^
[error] two errors found
```

Need an ACK about that move here and I'll do the thing.

**Comment by wsargent** [comment]
@ktoso the only concern I have is about reactive platform maintainence.  If it's not a factor in Akka going forwards and we can start using JDK 1.8 APIs, then great, but I don't know that that's the case.

**Comment by ktoso** [comment]
And just to make sure: Play is not yet using this, so for maintenance of JDK6/7 compatible Play, likely fixes would be in Play's codebase...

I am not sure how much of ssl-config is possible to rip out if we declare JDK8 – there's things like setting the hotname verification manually that we don't need to do on 8, right? Some of the algorithm defaults are also better – I guess it'd open up the door for lots of cleaning up here right?

I'll check with our team tomorrow if I'm not missing anything (technically we do have 2.0.x of Akka HTTP, so the JDK6 ssl-config is in there...), we'll think about it more then.

**Comment by ktoso** [comment]
Btw, in any case – I'm working on making all the config classes easier to binary compatible-evolve (case classes hurt that aspect :-()

**Comment by ktoso** [comment]
Ok, we'll go ahead with 2 branches here so we're able to maintain 0.1.x if need be.

**Pull request merged**